src/config/backend.ts

```diff
@@ -424,3 +439,218 @@ export class ExtlinuxConfigBackend extends DeviceConfigBackend {
+		constants.rootMountPoint,
+		'/sys/kernel/config',
+	);
+	private readonly ConfigVarNamePrefix = `${constants.hostConfigVarPrefix}CONFIGFS_`;
+
+	// supported backend for the following device types...
+	public static readonly SupportedDeviceTypes = ['up-board'];
+
+	private stripPrefix(name: string): string {
+		return name.substr(this.ConfigVarNamePrefix.length);
+	}
+
+	private async listLoadedAcpiTables(): Promise<string[]> {
+		const acpiTablesDir = path.join(this.ConfigfsMountPoint, '/acpi/table');
+		return await fs.readdir(acpiTablesDir);
+	}
+
+	private async loadAML(aml: string): Promise<boolean> {
+		if (!aml) {
+			return false;
+		}
+
+		const amlSrcPath = path.join(this.SystemAmlFiles, `${aml}.aml`);
+		// log to system log if the AML doesn't exist...
+		if (!(await fs.exists(amlSrcPath))) {
+			log.error(`Missing AML for \'${aml}\'. Unable to load.`);
+			if (this.options.logger) {
+				this.options.logger.logSystemMessage(
+					`Missing AML for \'${aml}\'. Unable to load.`,
+					{ aml, path: amlSrcPath },
+					'Load AML error',
+					false,
+				);
+			}
+			return false;
+		}
+
+		const amlDstPath = path.join(this.ConfigfsMountPoint, '/acpi/table', aml);
+		try {
+			const loadedTables = await this.listLoadedAcpiTables();
+
+			if (loadedTables.indexOf(aml) < 0) {
+				await fs.mkdir(amlDstPath);
+			}
+
+			log.info(`Loading AML ${aml}`);
+			await child_process.exec(
+				`cat ${amlSrcPath} > ${path.join(amlDstPath, 'aml')}`,
+			);
+
+			const [oemId, oemTableId, oemRevision] = await Bluebird.all([
+				fs.readFile(path.join(amlDstPath, 'oem_id'), 'utf8'),
+				fs.readFile(path.join(amlDstPath, 'oem_table_id'), 'utf8'),
+				fs.readFile(path.join(amlDstPath, 'oem_revision'), 'utf8'),
+			]);
+
+			log.info(
+				`AML: ${oemId.trim()} ${oemTableId.trim()} (Rev ${oemRevision.trim()})`,
+			);
+		} catch (e) {
+			log.error(e);
+		}
+		return true;
+	}
+
+	private async readConfigJSON(): Promise<ConfigfsConfig> {
+		// if we don't yet have a config file, just return an empty result...
+		if (!(await fs.exists(this.ConfigFilePath))) {
+			log.info('Empty ConfigFS config file');
+			return {};
+		}
+
+		// read the config file...
+		try {
+			const content = await fs.readFile(this.ConfigFilePath, 'utf8');
+			return JSON.parse(content);
+		} catch (err) {
+			log.error('Unable to deserialise ConfigFS configuration.', err);
+			return {};
+		}
+	}
+
+	private async writeConfigJSON(config: ConfigfsConfig): Promise<void> {
+		await remountAndWriteAtomic(this.ConfigFilePath, JSON.stringify(config));
+	}
+
+	private async loadConfiguredSsdt(config: ConfigfsConfig): Promise<void> {
+		if (config['ssdt'] !== undefined && _.isArray(config['ssdt'])) {
+			log.info('Loading configured SSDTs');
+			await Bluebird.mapSeries(config['ssdt'], aml => this.loadAML(aml));
+		}
+	}
+
+	public async initialise(
+		opts: BackendOptions,
+	): Promise<ConfigfsConfigBackend> {
+		try {
+			super.initialise(opts);
```

Comment:
This works because whilst `super.initialise` returns a promise it actually does all it's work synchronously, if we need to keep it as a promise returning function though then we should make sure to `await` this

Reply:
yep - I spotted this before I read this comment, good spot 👍